src/aspire/abinitio/commonline_c2.py

```diff
@@ -0,0 +1,382 @@
+import logging
+
+import numpy as np
+from scipy.linalg import eigh
+
+from aspire.abinitio import CLSymmetryC3C4
+from aspire.utils import J_conjugate, Rotation, all_pairs
+
+logger = logging.getLogger(__name__)
+
+
+class CLSymmetryC2(CLSymmetryC3C4):
+    """
+    Define a class to estimate 3D orientations using common lines methods for molecules with C2 cyclic symmetry.
+
+    The related publications are listed below:
+    X. Cheng,
+    Random Matrices in High-dimensional Data Analysis,
+    PhD thesis, Princeton University, (2013).
+
+    G. Pragier and Y. Shkolnisky,
+    A Common Lines Approach for Abinitio Modeling of Cyclically Symmetric Molecules,
+    Inverse Problems, 35, 124005, (2019).
+
+    Y. Shkolnisky, and A. Singer,
+    Viewing Direction Estimation in Cryo-EM Using Synchronization,
+    SIAM J. Imaging Sciences, 5, 1088-1110 (2012).
+
+    A. Singer, R. R. Coifman, F. J. Sigworth, D. W. Chester, Y. Shkolnisky,
+    Detecting Consistent Common Lines in Cryo-EM by Voting,
+    Journal of Structural Biology, 169, 312-322 (2010).
+    """
+
+    def __init__(
+        self,
+        src,
+        n_rad=None,
+        n_theta=None,
+        max_shift=0.15,
+        shift_step=1,
+        epsilon=1e-3,
+        max_iters=1000,
+        degree_res=1,
+        min_dist_cls=25,
+        seed=None,
+    ):
+        """
+        Initialize object for estimating 3D orientations for molecules with C2 symmetry.
+
+        :param src: The source object of 2D denoised or class-averaged images with metadata
+        :param n_rad: The number of points in the radial direction
+        :param n_theta: The number of points in the theta direction
+        :param max_shift: Maximum range for shifts as a proportion of resolution. Default = 0.15.
+        :param shift_step: Resolution of shift estimation in pixels. Default = 1 pixel.
+        :param epsilon: Tolerance for the power method.
+        :param max_iter: Maximum iterations for the power method.
+        :param degree_res: Degree resolution for estimating in-plane rotations.
+        :param min_dist_cls: Minimum distance between mutual common-lines. Default = 25 degrees.
+        :param seed: Optional seed for RNG.
+        """
+        super().__init__(
+            src,
+            symmetry="C2",
```

Comment:
We might be able to do away with this parameter altogether now that our sources have a `SymmetryGroup` property. Either way, I agree that we should do something different here. Since this will need to be done for all the common lines methods, I view this as part of the cleanup/reorganization of the common lines classes that should take place after this PR.

Reply:
Agreed. Maybe make an issue.

Reply:
Added issue #975